### PR TITLE
bhyve: Support PCI INTx on arm64, build the affected devices and add a PL031 RTC

### PR DESCRIPTION
**Pull request by jrtc27**

### CODE REVIEW

File: usr.sbin/bhyve/amd64/pci_irq_md.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+pci_irq_init_irq(struct pci_irq *irq)
+{
+	irq->pirq_pin = 0;
+	irq->ioapic_irq = 0;
+}
+
+static inline uint8_t
+pci_irq_intline(struct pci_irq *irq)
+{
+	return (pirq_irq(irq->pirq_pin));
```
Comment (bsdjhb) [inline usr.sbin/bhyve/amd64/pci_irq_md.h:56]: Hmm, I can't find the implementation of `pirq_irq` in this patch for amd64 now, seems like this wouldn't compile on amd64?

**Review by bsdjhb — COMMENTED** [review]
Generally looks good to me.  Do want to coordinate with markj on getting these upstreamed eventually.

File: usr.sbin/bhyve/amd64/pci_irq_md.h (hunk 1 above)
Comment (bsdjhb) [reply usr.sbin/bhyve/amd64/pci_irq_md.h:56]: (And yet it compiles in CI so must be there somewhere?)

Comment (jrtc27) [reply usr.sbin/bhyve/amd64/pci_irq_md.h:56]: Hmmm not sure what's going on there

Comment (jrtc27) [reply usr.sbin/bhyve/amd64/pci_irq_md.h:56]:
```        if self.crosscompile_target.is_any_x86():
            # seems to be missing some include paths which appears to work on freebsd
            self.make_args.set_with_options(BHYVE=False)
``` :(

Comment (jrtc27) [reply usr.sbin/bhyve/amd64/pci_irq_md.h:56]: Yeah, assuming that's the only thing I broke

Comment (bsdjhb) [reply usr.sbin/bhyve/amd64/pci_irq_md.h:57]: The updated version worked for me on my amd64 laptop, I tested an i386 VM with MSI disabled (so using INTx interrupts via the I/O APIC) for ahci0, and also with APIC disabled entirely, so using the PIR and 8259A PICs.

**Review by bsdjhb — APPROVED** [review]

**Pull request merged**